isotovideo

```diff
@@ -262,92 +246,33 @@ $backend = OpenQA::Isotovideo::Backend->new;
 
 spawn_debuggers;
 
-my $io_select = IO::Select->new();
-$io_select->add($testfd);
-$io_select->add($cmd_srv_fd);
-$io_select->add($backend->process->channel_out);
-
 # stop main loop as soon as one of the child processes terminates
-my $stop_loop = sub (@) { $loop = 0 if $loop; };
+my $stop_loop = sub (@) { $command_handler->loop(0) if $command_handler->loop; };
 $testprocess->once(collected => $stop_loop);
 $backend->process->once(collected => $stop_loop);
 $cmd_srv_process->once(collected => $stop_loop);
 
-# now we have everything, give the tests a go
-$testfd->write("GO\n");
-
 $command_handler = OpenQA::Isotovideo::CommandHandler->new(
     cmd_srv_fd => $cmd_srv_fd,
+    test_fd => $testfd,
     backend_fd => $backend->process->channel_in,
+    backend_out_fd => $backend->process->channel_out,
 );
 $command_handler->on(tests_done => sub (@) {
         CORE::close($testfd);
         $testfd = undef;
         stop_autotest;
-        $loop = 0;
 });
-
-my ($last_check_seconds, $last_check_microseconds) = gettimeofday;
-sub _calc_check_delta () {
-    # an estimate of eternity
-    my $delta = $last_check_seconds ? tv_interval([$last_check_seconds, $last_check_microseconds], [gettimeofday]) : 100;
-    # sleep the remains of one second if $delta > 0
-    my $timeout = $delta > 0 ? 1 - $delta : 0;
-    $command_handler->timeout($timeout < 0 ? 0 : $timeout);
-    return $delta;
-}
-
-sub check_asserted_screen ($no_wait = undef) {
-    if ($no_wait) {
-        # prevent CPU overload by waiting at least a little bit
-        $command_handler->timeout(0.1);
-    }
-    else {
-        _calc_check_delta;
-        # come back later, avoid too often called function
-        return if $command_handler->timeout > 0.05;
-    }
-    ($last_check_seconds, $last_check_microseconds) = gettimeofday;
-    my $rsp = $bmwqemu::backend->_send_json({cmd => 'check_asserted_screen'}) || {};
-    # the test needs that information
-    $rsp->{tags} = $command_handler->tags;
-    if ($rsp->{found} || $rsp->{timeout}) {
-        myjsonrpc::send_json($testfd, {ret => $rsp});
-        $command_handler->clear_tags_and_timeout();
-    }
-    else {
-        _calc_check_delta unless $no_wait;
-    }
-}
+$command_handler->on(signal => sub ($event, $sig) {
```

Comment:
The event you're emitting is just called `sig` so this handler won't be executed.

Reply:
It looks still inconsistent. The signal is now called `signalhandler`.